python/src/iceberg/literals.py

```diff
@@ -0,0 +1,595 @@
+            return False
+
+        return self.value <= other.value
+
+    def __ge__(self, other):
+        if other is None:
+            return True
+
+        return self.value >= other.value
+
+    def __str__(self):
+        return '"' + self.value + '"'
+
+
+class UUIDLiteral(ComparableLiteral):
+    def __init__(self, value):
+        super(UUIDLiteral, self).__init__(value)
+
+    def to(self, type_var):
+        if isinstance(type_var, UUIDType):
+            return self
+
+
+class FixedLiteral(BaseLiteral):
+    def __init__(self, value):
+        super(FixedLiteral, self).__init__(value)
+
+    def to(self, type_var):
+        if isinstance(type_var, FixedType):
+            if len(self.value) == type_var.length:
+                return self
+        elif isinstance(type_var, BinaryType):
+            return BinaryLiteral(self.value)
+
+    def write_replace(self):
+        return FixedLiteralProxy(self.value)
+
+    def __eq__(self, other):
+        return self.value == other.value
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        if other is None:
+            return False
+
+        return self.value < other.value
+
+    def __gt__(self, other):
+        if other is None:
+            return True
+
+        return self.value > other.value
+
+    def __le__(self, other):
+        if other is None:
+            return False
+
+        return self.value <= other.value
+
+    def __ge__(self, other):
+        if other is None:
+            return True
+
+        return self.value >= other.value
+
+
+class BinaryLiteral(BaseLiteral):
+    def __init__(self, value):
+        super(BinaryLiteral, self).__init__(value)
+
+    def to(self, type_var):
+        if isinstance(type_var, FixedType):
+            if type_var.length == len(self.value):
+                return FixedLiteral(self.value)
+            return None
+        elif isinstance(type_var, BinaryType):
+            return self
+
+    def write_replace(self):
+        return BinaryLiteralProxy(self.value)
+
+    def __eq__(self, other):
+        return self.value == other.value
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        if other is None:
+            return False
+
+        return self.value < other.value
+
+    def __gt__(self, other):
+        if other is None:
+            return True
+
+        return self.value > other.value
+
+    def __le__(self, other):
+        if other is None:
+            return False
+
+        return self.value <= other.value
+
+    def __ge__(self, other):
+        if other is None:
+            return True
+
+        return self.value >= other.value
+
+
+class FixedLiteralProxy(object):
+    def __init__(self, buffer=None):
+        if buffer is not None:
+            self.bytes = list(buffer)
+
+    def read_resolve(self):
+        return FixedLiteral(self.bytes)
+
+
+class ConstantExpressionProxy(object):
```

Comment:
@TGooch44 (or anyone else)--sorry if this is obvious but what's the purpose of these "Proxy" classes?

Reply:
Since multiple reviewer suggested, I end up removing such code as it's not being used. Once we get needs and clarify, we can always add them later if needed. So resolve this